app/views/course/announcements/_form.html.slim

```diff
@@ -0,0 +1,12 @@
+= simple_form_for [@course, @announcement],
+                    url: (@announcement.new_record? ?  course_announcements_url(@course) : \
+                         course_announcement_url(@course, @announcement)) do |f|
```

Comment:
Gotta think of a better way to deal with this.

I'm leaning to overriding self.model_name. I _believe_ we can just override it in the Course model.